Modules/System Tests/SharePoint Connector/src/SharePointClientTest.Codeunit.al

```diff
@@ -0,0 +1,415 @@
+codeunit 132970 "SharePoint Client Test"
+{
+    Subtype = Test;
+
+    var
+        SharePointTestLibrary: Codeunit "SharePoint Test Library";
+        Assert: Codeunit "Library Assert";
+        Any: Codeunit Any;
+        DummySharePointAuthorization: Codeunit "Dummy SharePoint Authorization";
+        SharePointClient: Codeunit "SharePoint Client";
+        BaseUrl: Text;
+        IsInitialized: Boolean;
+
+    [Test]
+    procedure TestGetLists()
+    var
+        TempSharePointList: Record "SharePoint List" temporary;
+    begin
+        // [Scenario] GetLists operation succeds and records are returned
+        Initialize();
+
+        SharePointClient.GetLists(TempSharePointList);
+
+        Assert.AreEqual(2, TempSharePointList.Count(), 'Expected 2 records');
+        TempSharePointList.FindFirst();
+
+        Assert.AreEqual('100', TempSharePointList."Base Template", StrSubstNo('Different %1 value expected', TempSharePointList.FieldCaption("Base Template")));
+        Assert.AreEqual('0', TempSharePointList."Base Type", StrSubstNo('Different %1 value expected', TempSharePointList.FieldCaption("Base Type")));
+        Assert.AreEqual('2022-05-23T12:16:04Z', Format(TempSharePointList.Created, 0, 9), StrSubstNo('Different %1 value expected', TempSharePointList.FieldCaption("Created")));
+        Assert.AreEqual('My Test Documents', TempSharePointList.Description, StrSubstNo('Different %1 value expected', TempSharePointList.FieldCaption("Description")));
+        Assert.AreEqual('{854D7F21-1C6A-43AB-A081-20404894B449}', Format(TempSharePointList.Id), StrSubstNo('Different %1 value expected', TempSharePointList.FieldCaption("Id")));
+        Assert.AreEqual(false, TempSharePointList."Is Catalog", StrSubstNo('Different %1 value expected', TempSharePointList.FieldCaption("Is Catalog")));
+        Assert.AreEqual('SP.Data.My_x0020_Test_x0020_DocumentsListItem', TempSharePointList."List Item Entity Type", StrSubstNo('Different %1 value expected', TempSharePointList.FieldCaption("List Item Entity Type")));
+        Assert.AreEqual('Web/Lists(guid''854d7f21-1c6a-43ab-a081-20404894b449'')', TempSharePointList.OdataEditLink, StrSubstNo('Different %1 value expected', TempSharePointList.FieldCaption("OdataEditLink")));
+        Assert.IsTrue(TempSharePointList.OdataId.EndsWith('_api/Web/Lists(guid''854d7f21-1c6a-43ab-a081-20404894b449'')'), StrSubstNo('Different %1 value expected', TempSharePointList.FieldCaption("OdataId")));
```

Comment:
I don't know what's the best way to do this, but maybe this guid, that is used inside the library and inside the Test Codeunit should be placed inside a helper function of the library and instead of copying the guid everywhere this dummyGuid procedure could be used instead. 
Then it would be more intuitive to create more test, as soon as the functionality is added.

Reply:
The same guid repeats a lot because all the tests are referring to the same list/folder. It's not constant across requests like the URL (where there is a dummy value helper - mostly to anonymize the data).

Right now creating a new test response involves grabbing the actual json returned by the server, pasting it to the library, and replacing URL with the dummy one. 
The less you have to tamper with the response the easier it is, and the more reliable the test.